Bump version: canghaicheng.ZhuoMianShiJie version from 1.0.5 to 1.0.9

Dragon1573 opened the pull request:
## What's changed?

- Switched to SourceForge download link for the installer.
- Updated installer URL and SHA256 hash.

## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #247375

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

![Image](https://github.com/user-attachments/assets/4d09d61b-cabf-4125-af73-3c4e2b592c96)

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/247832)

Dragon1573 committed:
Update manifest: canghaicheng.ZhuoMianShiJie version 1.0.5

- Switched to SourceForge download link for the installer.
- Updated installer URL and SHA256 hash.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

stephengillie commented:
The `PackageUrl` didn't lead to the new `InstallerUrl`. My laptop had difficulty rendering it:
![image](https://github.com/user-attachments/assets/e39c7678-56d1-4a4f-ad3e-231e75874ffe)

Dragon1573 commented:
> The `PackageUrl` didn't lead to the new `InstallerUrl`. My laptop had difficulty rendering it:

At the top-left corner, there's a "下载 (Download)" button. It can redirect to the project homepage on SourceForge.

![image](https://github.com/user-attachments/assets/d355ed97-77c4-4c9a-83ac-f628fddf0b2b)

Dragon1573 committed:
Update manifest: canghaicheng.ZhuoMianShiJie version 1.0.9

- Update the description to be more readable.
- Update the package URL to point to the SourceForge page.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>
